purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1387,29 +1387,6 @@ class Purchases internal constructor(
                     }
                 }
 
-                if (purchases.isEmpty()) {
-                    if (isDeprecatedProductChangeInProgress) {
-                        // Can happen if the product change is ProrationMode.DEFERRED
-                        invalidateCustomerInfoCache()
-                        getCustomerInfoWith { customerInfo ->
-                            deprecatedProductChangeListener?.let { callback ->
-                                dispatch {
-                                    callback.onCompleted(null, customerInfo)
-                                }
-                            }
-                        }
-                    } else {
-                        // the non-deprecated purchase(PurchaseParams, PurchaseCallback) flow doesn't allow for
-                        // DEFERRED, so this is an unexpected state. return an error
-                        val nullTransactionError = PurchasesError(
-                            PurchasesErrorCode.StoreProblemError,
-                            PurchaseStrings.NULL_TRANSACTION_ON_PURCHASE_ERROR
-                        )
-                        getAndClearAllPurchaseCallbacks().forEach { it.dispatch(nullTransactionError) }
-                    }
-                    return
-                }
```

Comment:
Entirely removed the special handling for the case where purchases are empty:

- It was only being handled when using the deprecated purchase methods.
- In the new purchase method callback, `StoreTransaction` is not nullable, so I cannot 
- Since Billing Library 5, the transactions list will contain the transaction for the old product. See [comment](https://github.com/RevenueCat/purchases-android/pull/985/#issuecomment-1535817218)

We *could* make the new purchase method callback have a nullable transaction parameter, and handle the case where `purchases.isEmpty()` for both old and new purchase methods. But supposedly that should not be happening in BL5.

Thoughts?

Reply:
Hmm in the `BillingWrapper` class we are still forwarding an empty list of purchases, even though it shouldn't happen in BC5. If that unexpected situation happened, we wouldn't call the callbacks after removing this if I'm not wrong, so it would basically get stuck.

I think we should call the error callback when this happens, at least until we are certain that this won't happen, and at that point, we should remove that code from BillingWrapper